### PR TITLE
WebGPURenderer: Particles example

### PR DESCRIPTION
**Pull request by sunag**
### Description

It was relatively easy to create particles using `NodeMaterial` system.
I created a fire example.

### Live

https://raw.githack.com/sunag/three.js/dev-webgpu-particles-v1/examples/webgpu_particles.html

image

### Commits

- [x] [WebGPU: particles example](https://github.com/mrdoob/three.js/commit/2d483cae662db5c210f62d2989d93aa08edf714d)
- [x] [WebGPUGeometries: update geometry if was disposed on renderer process](https://github.com/mrdoob/three.js/commit/aaf9a27984039731625b6e50ebe15c813acf2f49)
- [x] [TimerNode: add initial value for timer](https://github.com/mrdoob/three.js/commit/641737e6744278f06498c54f46e828e45d8bc27d)
- [x] [Add RotateUVNode](https://github.com/mrdoob/three.js/commit/bde2fad2b81caa65cc93ad8bf19f390bb66d3f54)
- [x] [SpriteNodeMaterial: move .positionNode to work like particles](https://github.com/mrdoob/three.js/commit/86906e75c87f77238b7787571c4a434ccb61c5be)
- [x] [WebGPURenderer: fix AdditiveBlending](https://github.com/mrdoob/three.js/commit/452de4688a6b8fdb219c23ceb1bc709834260f2d)
- [x] [RangeNode: reuse buffer used in others mesh/sprite](https://github.com/mrdoob/three.js/commit/e4bfb69ada4faa6347942d53cb6f03d5bc13d4a4)

### Possible issues

- [x] Nodes: add `64-bits` precision, long values can generate flickers
- [x] Nodes: Use `Mesh` instead of `Sprites` for not overload `GeometryBuffers` shared.
- [x] WebGPU: Create an example to check all `blendModes`.

*This contribution is funded by [Google via Igalia](https://igalia.com)*

### CODE REVIEW

File: examples/jsm/nodes/materials/SpriteNodeMaterial.js [hunk 1]
```diff
@@ -44,13 +44,7 @@ class SpriteNodeMaterial extends NodeMaterial {
 
 		let vertex = positionLocal;
```
Comment (LeviPesin) [inline examples/jsm/nodes/materials/SpriteNodeMaterial.js:45]: I think `vertex` should be made `const`?

File: examples/webgpu_particles.html [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+				// create particles
+
+				const smokeNodeMaterial = new Nodes.SpriteNodeMaterial();
+				smokeNodeMaterial.colorNode = mul( mix( color( 0xf27d0c ), smokeColor, min( mul( life, 2.5 ), 1 ) ), fakeLightEffect );
+				smokeNodeMaterial.opacityNode = opacityNode;
+				smokeNodeMaterial.positionNode = mul( offsetRange, lifeTime );
+				smokeNodeMaterial.scaleNode = mul( scaleRange, max( .3, lifeTime ) );
+				smokeNodeMaterial.depthWrite = false;
+				smokeNodeMaterial.depthTest = true;
```
Comment (LeviPesin) [inline examples/webgpu_particles.html:96]: `Material.depthTest` is already `true` by default, I think?

**Comment by sunag** [comment]
@mrdoob @Mugen87 Something that I can improve here?

**Review by Mugen87 — APPROVED** [review]
LGTM!

File: examples/webgpu_particles.html [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+				//
+
+				window.addEventListener( 'resize', onWindowResize );
+
+				// gui
+
+				const gui = new GUI();
+
+				gui.add( timer, 'scale', 0, 1, 0.01 );
```
Comment (mrdoob) [inline examples/webgpu_particles.html:151]: Nit:
```js
gui.add( timer, 'scale', 0, 1, 0.01 ).name( 'speed' );
```

**Pull request merged**